### PR TITLE
avifenc: Allow large images to be encoded

### PR DESCRIPTION
**Pull request by wantehchang**
Pass UINT32_MAX as the imageSizeLimit argument to avifReadImage().

Bug: https://github.com/AOMediaCodec/libavif/issues/2271

### CODE REVIEW

**Review by wantehchang — COMMENTED** [review]
Yannis: Please review this pull request carefully. I remember you added the `imageSizeLimit` parameter to `avifReadImage()` to avoid out-of-memory errors in fuzzing tests. Also,  `avifReadImage()` passes that parameter to `avifJPEGRead()` as the `sizeLimit` parameter, which is also used to limit the size of Exif and XMP metadata. So I don't fully understand the implications of this pull request.

File: apps/avifenc.c [hunk 1]
```diff
@@ -611,7 +611,7 @@ static avifBool avifInputReadImage(avifInput * input,
                                                             ignoreXMP,
                                                             allowChangingCicp,
                                                             ignoreGainMap,
-                                                            AVIF_DEFAULT_IMAGE_SIZE_LIMIT,
```
Comment (y-guyon) [inline apps/avifenc.c:614]: Should it also be changed at line 585?

**Review by y-guyon — APPROVED** [review]
I think this change is fine. Metadata >256MB makes little sense but this is not worth adding a parameter to `avifReadImage()`.

File: apps/avifenc.c (hunk 1 above)
Comment (wantehchang) [reply apps/avifenc.c:614]: Done. Thanks.

I found that `AVIF_DEFAULT_IMAGE_SIZE_LIMIT` is also used in two places in `avifgainmaputil`. I didn't look at them to see if either of them should be changed to `UINT32_MAX`.

**Review by wantehchang — COMMENTED** [review]
Thanks for the review. I also added a changelog entry. PTAL.

**Review by y-guyon — APPROVED** [review]

**Pull request merged**